docs/vagrant-spk/debugging.md

```diff
@@ -0,0 +1,245 @@
+  executing `rm -f .sandstorm/sandstorm-files.list` instead.
+
+If you do not do this, you would add debugging tools to the package that your users receive,
+massively increasing the size of the SPK download.
+
+## Tips on using the shell effectively
+
+### Understanding the filesystem layout
+
+The filesystem available in this shell is the same as the filesystem available within your grain.
+You can find information on this in the `.sandstorm/sandstorm-pkgdef.capnp` file. As a reminder,
+consider the following.
+
+- `/` (read-only) contains all the files that could hypothetically go into your package. The bash process
+  launched by `vagrant-spk enter-grain`shell starts in `/`.
+
+- `/var` (read-write) contains the writable state for your app. This maps to
+  `/opt/sandstorm/var/sandstorm/grains/{{grainId}}/sandbox` outside the grain.
+
+If your `.sandstorm/sandstorm-pkgdef.capnp` contains a `sourcePath = "/"` line, which is the default
+for all vagrant-spk platform stacks except the Meteor stack, then the full contents of the Vagrant
+base box are available. If not, read below on how to add that.
+
+`vagrant-spk` configures `.sandstorm/Vagrantfile` so that `/opt/app` (read-only within the grain) is
+a shared folder to your host operating system, typically containing your app's source code. If your
+app contains command-line tools, then you probably need to run this command before you can access
+those tools.
+
+```bash
+cd /opt/app
+```
+
+### Overcoming missing command-line tools like ls
+
+Some apps, especially those using Meteor platform stack for vagrant-spk, do not configure `sourcePath = "/"`,
+so if you try to run some commands, you will get an error. For example:
+
+```bash
+$ ls
+bash: ls: command not found
+```
+
+To work around this, you have three options.
+
+**Option 1. Use debugging tools that are available.** For the Meteor stack in particular, you can
+choose to run `node` instead, which provides a more Meteor-like experience while debugging. Here is
+how you can launch node from `vagrant-spk enter-grain`.
+
+```bash
+$ vagrant-spk enter-grain
+$ /bin/node
+>   # you can now run whatever Javascript you wish
+```
+
+**Option 2. Inject the specific tools you need.** One good way to do this is to inject busybox into
+the grain. To do that, you can run the following.
+
+```bash
+$ vagrant-spk vm ssh
+$ sudo cp /bin/busybox /opt/sandstorm/var/sandstorm/grains/{{grainId}}/sandbox
+```
+
+Now, when you enter the grain, you can use busybox's bundled versions of ls, cp, and other typical
+tools.
+
+```bash
+$ vagrant-spk enter-grain
+$ ls
+bash: ls: command not found
+bash-4.3$ /var/busybox sh
+$ ls  # will work now, since it comes from busybox
+```
+
+**Option 3. Enable filesystem tracing for this app.** In `.sandstorm/sandstorm-pkgdef.capnp`, look for this
+line, which disables tracing.
+
+```bash
+alwaysInclude = [ "." ] ,
+```
+
+Replace it with this line, which enables tracing.
```

Comment:
This isn't quite right - `alwaysInclude` and tracing being enabled are orthogonal.  See [sandstorm-test-python](https://github.com/zarvox/sandstorm-test-python/blob/master/.sandstorm/sandstorm-pkgdef.capnp) for an example which uses both tracing and `alwaysInclude`.

It's `fileList` that needs to be provided to enable tracing.  Combined with the `sourcePath = "/"` below, that allows mapping in any file on the system.

Suggest reworking this section.